app/models/ability.rb

```diff
@@ -0,0 +1,51 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    user ||= User.new # guest user (not logged in)
+    if user.admin?
+      can :manage, :all
+    else
+      can :read, :all
+      can :create, [Question, Answer, Comment]
+
+      can [:update, :destroy], Question do |question|
+        user.id == question.user_id 
```

Comment:
there is another way of defining ability in cancancan without blocks. Look for that as well.